### ISSUE TITLE
Entire Tweet, Location, and Language in "get_all_tweets"

### ISSUE THREAD

**Issue by JijoC-13** [cjbarrie/academictwitteR#108]
**Entire Tweet, Location, and Language**
The retrieved tweets by the code "get_all_tweets" are not complete. It only covers up-to what the "Keywords" of the query is.
The Country is specified but not seen in the retrieved data. And language are mixed.

**To Reproduce**
1. devtools::install_github("cjbarrie/academictwitteR", build_vignettes = TRUE)
bearer token <- "##########"
2. The load the package
3. The run the following code:
tweets <-get_all_tweets("Autonomous Driving OR self driving cars OR autopilot",
                                      "2020-06-01T00:00:00Z",
                                      "2020-07-01T00:00:00Z",is_retweet = FALSE, lang = "en",country = "US",
                                      bearer_token, data_path = "data/", bind_tweets = FALSE)
I have attached the screenshot of the data. It is not an error but not able to access the required information which was available with rtweet for the standard package.
**Expected behavior**
I expected that I would get all the words of the tweets as I have to do a sentiment analysis. And also, the language is not specific and the location cannot be confirmed. If the instruction to give only english is not accepted, then I guess the country variable is also not considered.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: iOS
 - Version 10.15.6

**Additional context**
Upon inspection I was able to find that the tweets are still able to access if there is only one keyword. If there more than one separated with "OR" then none of the mentioned parameters are fulfilled, and the tweets are not complete. The text stop after a few words where the "key query" has been fullfilled.
![Screenshot 2021-06-08 at 15 26 32](https://user-images.githubusercontent.com/85554275/121193475-2bb25f80-c86e-11eb-98c6-142974228ae6.png)
![Screenshot 2021-06-08 at 15 25 26](https://user-images.githubusercontent.com/85554275/121193496-2ead5000-c86e-11eb-8c27-396ebb956d56.png)
![Screenshot 2021-06-08 at 15 16 56](https://user-images.githubusercontent.com/85554275/121193498-2f45e680-c86e-11eb-99d5-794b2caa0524.png)
![Screenshot 2021-06-08 at 15 30 22](https://user-images.githubusercontent.com/85554275/121193881-7e8c1700-c86e-11eb-9fca-eaf5c1a5a839.png)

NOTE: The screenshots do not contain the entire data. Only enough to show the problem

**Comment by chainsawriot** [comment]
ref ropensci/rtweet#575

**Comment by andersolarsson** [comment]
Hello, 

First of all, thank you for a very good package! As for my question, a problem similar to the one mentioned by JijoC-13 above is present also for "get_user_tweets" - tweet text appears to be shortened and I can't figure out if there is some sort of logic to this abbreviation. The link above - [ropensci/rtweet#575](https://github.com/ropensci/rtweet/issues/575) - suggest that tweet_mode = "extended" could be used somehow, however academictwitteR does not seem to recognize this argument. 

Am I missing something here, or is this currently an unresolved issue - how to get the full tweet text using academictwitteR ?

**Comment by chainsawriot** [comment]
@andersolarsson Those are actually verbatim retweets. By default (and currently there is no way to fix in V2 API), they are trimmed. Actually, the "full text" of the source tweets are available from the saved json files (if you have specified `data_path`). It is still determining how to deliver the so-called 'tidy' data frame (#112). But please stay tuned. Below is a little preview.

``` r
require(academictwitteR)
#> Loading required package: academictwitteR
require(tidyverse)
#> Loading required package: tidyverse
random_path <- academictwitteR:::.gen_random_dir()
test <- get_all_tweets("#standwithhongkong", data_path = random_path, 
                       start_tweets = "2021-06-21T00:00:00Z", 
                       end_tweets = "2021-06-21T10:00:00Z", verbose = FALSE)

test$text[1:3]
#> [1] "#StandWithHongKong https://t.co/pGonWTd3We"                                                                                                      
#> [2] "RT @GordonGChang: The people in #HongKong are resisting. Let’s pitch in against our common enemy, #China’s regime. #StandWithHongKong"           
#> [3] "RT @chjackie797: From 2013. Still applicable in 2021. How democracy has progress in Hong Kong\n\n#StandWithHongKong \n#AppleDaily \n#PressFreed…"
## files containing includes
files <- list.files(random_path, pattern = "^users_", full.names = TRUE)
includes_section <- jsonlite::read_json(files, simplifyVector = TRUE)
source_tweets <- data.frame(source_id = includes_section$tweets$id, source_text = includes_section$tweets$text)

get_id <- function(x) {
  if (is.null(x)) {
    return(NA)
  }
  x$id[1]
}

source_id <- purrr::map_chr(test$referenced_tweets, get_id)

tibble(text = test$text, source_id = source_id) %>% 
  left_join(source_tweets, by = "source_id")
#> # A tibble: 100 x 3
#>    text                           source_id    source_text                      
#>    <chr>                          <chr>        <chr>                            
#>  1 "#StandWithHongKong https://t… 14068912572… "One of the largest circulating …
#>  2 "RT @GordonGChang: The people… 14067963147… "The people in #HongKong are res…
#>  3 "RT @chjackie797: From 2013. … 14068453938… "From 2013. Still applicable in …
#>  4 "RT @GordonGChang: The people… 14067963147… "The people in #HongKong are res…
#>  5 "RT @chjackie797: The latest … 14068839531… "The latest news from Next Media…
#>  6 "RT @the852spirit: 1100: 🚨🚨… 14068086786… "1100: 🚨🚨🚨 BRB . Handling som…
#>  7 "RT @KokdamonLam: #AppleDaily… 14069121634… "#AppleDaily #StandWithHongKong …
#>  8 "RT @GordonGChang: The people… 14067963147… "The people in #HongKong are res…
#>  9 "RT @GordonGChang: The people… 14067963147… "The people in #HongKong are res…
#> 10 "#StandWithHongKong #AppleDai… <NA>          <NA>                            
#> # … with 90 more rows
```

<sup>Created on 2021-06-21 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by andersolarsson** [comment]
Hello again,

Thank you for reply. Perhaps I should clarify what I  am trying to do. The code below:

_get_user_tweets(users = "account_name_here",
                start_tweets = "2009-01-01T00:00:00Z",
                end_tweets = "2014-12-31T00:00:00Z",
                bearer_token = bearer_token,
                data_path = "../output/",
                bind_tweets = TRUE)_

… is my attempt to get all posts from a specified account name. If I can somehow get just the _created_at_ column along with the full _text_ and the various _public_metrics_-columns (e.g. _public_metrics.like.count_, _public_metrics.reply.count_ et.c.), that would solve what I'm working on now. I take it from your reply that a tidy format is of _academictwitteR_ data is being worked on (which sounds amazing) but is there  a way for me to get the mentioned variables in a data.frame format? Thanks

**Comment by chainsawriot** [comment]
@andersolarsson is it what you want?

``` r
require(academictwitteR)
#> Loading required package: academictwitteR
require(tidyverse)
#> Loading required package: tidyverse
random_path <- academictwitteR:::.gen_random_dir()
test <- get_all_tweets(users = "katypearce", data_path = random_path, 
                       start_tweets = "2021-06-21T00:00:00Z", 
                       end_tweets = "2021-06-21T10:00:00Z", verbose = FALSE)

test
#>   author_id possibly_sensitive    entities.mentions
#> 1  93782410              FALSE 3, 17, Livingstone_S
#> 2  93782410              FALSE                 NULL
#> 3  93782410              FALSE   0, 12, jeanburgess
#> 4  93782410              FALSE    0, 11, profknowak
#> 5  93782410              FALSE             0, 3, UW
#>           entities.annotations               referenced_tweets          source
#> 1                         NULL  retweeted, 1406633438266941445 Twitter Web App
#> 2                         NULL                            NULL Twitter Web App
#> 3 26, 29, 0.5926, Person, Burr replied_to, 1406793556572852227 Twitter Web App
#> 4                         NULL replied_to, 1406785045608677381 Twitter Web App
#> 5                         NULL replied_to, 1406763823802036230 Twitter Web App
#>   public_metrics.retweet_count public_metrics.reply_count
#> 1                           31                          0
#> 2                            0                          0
#> 3                            0                          2
#> 4                            0                          0
#> 5                            0                          0
#>   public_metrics.like_count public_metrics.quote_count               created_at
#> 1                         0                          0 2021-06-21T06:20:46.000Z
#> 2                         4                          0 2021-06-21T05:04:43.000Z
#> 3                         0                          0 2021-06-21T03:50:17.000Z
#> 4                         0                          0 2021-06-21T01:33:33.000Z
#> 5                         0                          0 2021-06-21T00:02:33.000Z
#>                    id
#> 1 1406859620744777728
#> 2 1406840484048216064
#> 3 1406821751820931072
#> 4 1406787341365039105
#> 5 1406764440473665536
#>                                                                                                                                           text
#> 1 RT @Livingstone_S: “This is the first time we’ve had a society in which almost by default, everything is recorded and shared and aggregated…
#> 2      The best part about being allergic to insect stings is being able to have a good reason to make someone else remove a nest or whatever.
#> 3                                                                         @jeanburgess Incorrect is Burr-Guess right? It is a smoother Burjis?
#> 4                                                                                                                            @profknowak True.
#> 5                                                                                                          @UW Thanks but isn't it closed now?
#>   lang     conversation_id in_reply_to_user_id
#> 1   en 1406859620744777728                <NA>
#> 2   en 1406840484048216064                <NA>
#> 3   en 1406793556572852227            18202677
#> 4   en 1406759806526840835          3903374964
#> 5   en 1406763823802036230            27103822

test %>% select(created_at, text) %>% bind_cols(test$public_metrics)
#>                 created_at
#> 1 2021-06-21T06:20:46.000Z
#> 2 2021-06-21T05:04:43.000Z
#> 3 2021-06-21T03:50:17.000Z
#> 4 2021-06-21T01:33:33.000Z
#> 5 2021-06-21T00:02:33.000Z
#>                                                                                                                                           text
#> 1 RT @Livingstone_S: “This is the first time we’ve had a society in which almost by default, everything is recorded and shared and aggregated…
#> 2      The best part about being allergic to insect stings is being able to have a good reason to make someone else remove a nest or whatever.
#> 3                                                                         @jeanburgess Incorrect is Burr-Guess right? It is a smoother Burjis?
#> 4                                                                                                                            @profknowak True.
#> 5                                                                                                          @UW Thanks but isn't it closed now?
#>   retweet_count reply_count like_count quote_count
#> 1            31           0          0           0
#> 2             0           0          4           0
#> 3             0           2          0           0
#> 4             0           0          0           0
#> 5             0           0          0           0
```

<sup>Created on 2021-06-21 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by andersolarsson** [comment]
Hello again, 

So in my installation of _academictwitteR_, _get_all_tweets()_ does not seem to have a _users_ argument. Trying:

test <- get_user_tweets(users = "sdriks", # actual user name tried
                                         start_tweets = "2020-06-21T00:00:00Z", 
                                         end_tweets = "2021-06-21T10:00:00Z", 
                                         verbose = FALSE,
                                         bearer_token = bearer_token)

… but this does not solve the issue  with the truncated text. If I understand you correctly, a solution to get the full text of the tweet is in the works? Thank you very much for your assistance so far!

**Comment by chainsawriot** [comment]
The @andersolarsson 's question and part of @JijoC-13 's question are solved in v0.2 (usernames and full text) with `bind_tweets(output_format = "tidy")`. The language filtering problem is actually the [deficiency of the Twitter API](https://twittercommunity.com/t/full-archive-search-by-language-pt-but-retrieve-other-languages/153116). We can't solve.
